bluemira/builders/EUDEMO/reactor.py

```diff
@@ -76,9 +81,10 @@ def run(self) -> Component:
         component.add_child(self.build_vacuum_vessel(component, ivc_boundary))
         component.add_child(self.build_divertor(component, divertor_face))
         component.add_child(self.build_blanket(component, blanket_face))
+        component.add_child(self.build_VV_thermal_shield(component))
         component.add_child(self.build_TF_coils(component))
         component.add_child(self.build_PF_coils(component))
-        component.add_child(self.build_thermal_shield(component))
+        self.build_cryo_thermal_shield(component)
```

Comment:
I would probably prefer the consistency of the `build_xxx` methods returning what they're building, rather than assigning components within the method. Is this possible here?

Reply:
need to get this in... we need to agree on the structure of the tree as a whole for the reactor. I understand that keeping the consistency would be nice.

Reply:
My thought was to create the parent `THERMAL_SHIELD` component `run`, then add the cryro/vacuum vessel parts from the returns values of the design stages.

```python
thermal_shield = Component(self.THERMAL_SHIELD)
component.add_child(thermal_shield)
...
thermal_shield.add_child(self.build_VV_thermal_shield)
...
thermal_shield.add_child(self.build_cryo_thermal_shield)
```

This way we keep the design stages consistent, and `run` is still the only method that's adding components to the tree.

Is what I've written equivalent to what you've got, or have I missed something?

Reply:
Yep, agreed

Reply:
https://github.com/Fusion-Power-Plant-Framework/bluemira/issues/1047